spec/pid_segment_spec.rb

```diff
@@ -2,77 +2,94 @@
+      expect(filled_pid.country_code).to eq "ZZ"
     end
+  end
 
-    it "sets values correctly" do
-      pid = HL7::Message::Segment::PID.new @base
-      expect(pid.set_id).to eq "1"
-      expect(pid.patient_id).to eq ""
-      expect(pid.patient_id_list).to eq "333"
-      expect(pid.alt_patient_id).to eq ""
-      expect(pid.patient_name).to eq "LastName^FirstName^MiddleInitial^SR^NickName"
-      expect(pid.mother_maiden_name).to eq ""
-      expect(pid.patient_dob).to eq "19760228"
-      expect(pid.admin_sex).to eq "F"
-      expect(pid.patient_alias).to eq ""
-      expect(pid.race).to eq "2106-3^White^HL70005^CAUC^Caucasian^L"
-      expect(pid.address).to eq ""
-      expect(pid.county_code).to eq "AA"
-      expect(pid.phone_home).to eq ""
-      expect(pid.phone_business).to eq ""
-      expect(pid.primary_language).to eq ""
-      expect(pid.marital_status).to eq ""
-      expect(pid.religion).to eq ""
-      expect(pid.account_number).to eq "555.55"
-      expect(pid.social_security_num).to eq "012345678"
-      expect(pid.driver_license_num).to eq ""
-      expect(pid.mothers_id).to eq ""
-      expect(pid.ethnic_group).to eq ""
-      expect(pid.birthplace).to eq ""
-      expect(pid.multi_birth).to eq ""
-      expect(pid.birth_order).to eq ""
-      expect(pid.citizenship).to eq ""
-      expect(pid.vet_status).to eq ""
-      expect(pid.nationality).to eq ""
-      expect(pid.death_date).to eq "201011110924-0700"
-      expect(pid.death_indicator).to eq "Y"
-      expect(pid.id_unknown_indicator).to eq ""
-      expect(pid.id_readability_code).to eq ""
-      expect(pid.last_update_date).to eq ""
-      expect(pid.last_update_facility).to eq ""
-      expect(pid.species_code).to eq ""
-      expect(pid.breed_code).to eq ""
-      expect(pid.strain).to eq ""
-      expect(pid.production_class_code).to eq ""
-      expect(pid.tribal_citizenship).to eq ""
+  describe '#id_readability_code='  do
+    it "aliases the id_reliability_code field as id_readability_code for backward compatibility" do
+      expect(filled_pid.id_readability_code).to eq "UA"
+
+      filled_pid.id_readability_code = "AL"
+      expect(filled_pid.id_reliability_code).to eq "AL"
     end
+  end
 
-    it "aliases the county_code field as country_code for backwards compatibility" do
-      pid = HL7::Message::Segment::PID.new @base
-      expect(pid.country_code).to eq "AA"
+  describe '#admin_sex=' do
+    context 'when admin_sex is filled with an invalid value' do
+      it 'raises an InvalidDataError' do
+        expect do
+          ["TEST", "A", 1, 2].each do |x|
+            filled_pid.admin_sex = x
+          end
+        end.to raise_error(HL7::InvalidDataError)
+      end
+    end
 
-      pid.country_code = "ZZ"
-      expect(pid.country_code).to eq "ZZ"
+    context 'when admin_sex is filled with a valid value' do
+      it 'does not raise any error' do
+        expect do
+          vals = %w[F M O U A N C] + [ nil ]
```

Comment:
Not familiar with this fiel, but I don't see `C` listed - https://hl7-definition.caristix.com/v2/HL7v2.6/Tables/0001

Reply:
Indeed, I didn't check this content against the doc but will clean it in a later commit